vrrpsyncd/vrrpsync.cpp

```diff
@@ -0,0 +1,640 @@
+        mac.erase(0, pos + delimiter.length());
+	}
+    token = mac.substr(0, pos);
+	ll_addr[idx] = (uint8_t)std::stoi(token, 0, 16);
+
+	/*Generate LL based interface mac*/
+	l3_addr.s6_addr[0] = 0xfe;
+	l3_addr.s6_addr[1] = 0x80;
+	l3_addr.s6_addr16[1] = 0;
+	l3_addr.s6_addr32[1] = 0;
+	l3_addr.s6_addr[8] = ll_addr[0] | 0x02;
+	l3_addr.s6_addr[9] = ll_addr[1];
+	l3_addr.s6_addr[10] = ll_addr[2];
+	l3_addr.s6_addr[11] = 0xff;
+	l3_addr.s6_addr[12] = 0xfe;
+	l3_addr.s6_addr[13] = ll_addr[3];
+	l3_addr.s6_addr[14] = ll_addr[4];
+	l3_addr.s6_addr[15] = ll_addr[5];
+
+	char v6addr[50];
+	inet_ntop(AF_INET6, &l3_addr, v6addr, INET6_ADDRSTRLEN);
+	std::string llstr(v6addr);
+    llstr = llstr + "/64";
+
+    ipv6ll = llstr;
+
+    return true;
+}
+
+bool VrrpSync::isAutoGeneratedIPv6LL(const string& vip, const string& ifname)
+{
+    if (vip.compare(0, strlen("fe80"), "fe80"))
+    {
+        return false;
+    }
+
+    /*Check if ll matches with IPv6 LL generated from system mac*/
+    string systemIntf = "eth0";
+    string systemIPv6LL;
+    if (getAutoIPv6LL(systemIntf, systemIPv6LL) == false)
+    {
+        SWSS_LOG_ERROR("Failed to get System mac");
+    }
+
+    /*Check if ll matches with IPv6 LL generated from interface mac
+     * This is necessary since in case of MCLAG, 
+     * Vlan mac changes to Active node System MAC*/
+    string intfIPv6LL;
+    if (getAutoIPv6LL(ifname, intfIPv6LL) == false)
+    {
+        SWSS_LOG_ERROR("Failed to get Interface MAC");
+    }
+
+    SWSS_LOG_NOTICE("VIP: %s Auto System LL %s, Auto Intf LL: %s",
+            vip.c_str(), systemIPv6LL.c_str(), intfIPv6LL.c_str());
+
+	if ((systemIPv6LL == vip) || (intfIPv6LL == vip))
+        return true;
+
+    return false;
+}
+
+
+void VrrpSync::VrrpUpdateNbr(string &ifname, int afi, string &vip, string &op)
+{
+    string cmd, res;
+    int ret;
+    string afi_str = (afi == AF_INET6)? "-6": "";
+
+    cmd = "ip " + afi_str + " neigh flush " + vip + " dev " + ifname;
+
+    ret = swss::exec(cmd, res);
+    if (ret)
+    {
+        SWSS_LOG_ERROR("Command '%s' failed with rc %d", cmd.c_str(), ret);
+    }
+}
+
+
+void VrrpSync::VrrpUpdateVipNbr(string &macVlanIf, string &ifname, int afi, string &vip, bool del)
+{
+    string op = del? "del": "replace";
+
+    VrrpUpdateNbr(ifname, afi, vip, op);
+    VrrpUpdateNbr(macVlanIf, afi, vip, op);
+}
+
+
+void VrrpSync::VrrpDbUpdate(string &macVlanIf, int ifindex, string &ifname, int afi, string &vip, string &vmac, bool del)
+{
+    string key;
+    string family;
+    string len;
+    std::vector<FieldValueTuple> vec;
+
+    if (afi == AF_INET6)
+    {
+        family = "ipv6";
+        len = "/128";        
+        if (isAutoGeneratedIPv6LL(vip, ifname))
+        {
+            SWSS_LOG_DEBUG("Skip adding autogenerated IPv6 LL ifname: %s, VIP: %s",
+                    macVlanIf.c_str(), vip.c_str());
+            return;
+        }
+    }
+    else
+    {
+        family = "ipv4";
```

Comment:
It doesn't seem to be using while creating a key for VRRP_TABLE.

Reply:
Will remove the unused variable 'family'.